src/test/Makefile.inc

```diff
@@ -323,7 +323,20 @@ CFLAGS += $(COMMON_FLAGS)
 ifneq ($(USING_JEMALLOC_HEADERS),y)
 CFLAGS += -Wsign-conversion
 endif
-
+ifeq ($(call check_compiler, icc), n)
```

Comment:
build out the list of flags once, and use it for every test.
The way you've done it now will negatively impact build time.